src/ADC.c

```diff
@@ -1,20 +1,109 @@
+    ADCON0bits.ADON=0;      //periferico  apagado, para ahorrar energia
+    
+    
+            
+    
+    
+}
+float valorPropano(void) {
+    
+    //convercion ADC 
+    
+    ADCON0bits.CHS=0;       //seleccion de canal 0
+    ADCON0bits.ADON=1;      //encender el periferico ADC
+    __delay_us(10);          //espera requerida por el ADC
+    ADCON0bits.GO=1;        //Inicia Medicion en AD0    
+    while(ADCON0bits.GO);      //Se queda esperando hasta que acabe la conversion
+    float R_adc=ADRES;
+    ADCON0bits.ADON=0;      //apagar el periferico ADC para ahorrar energia
+    
+    //escalamiento de las unidades
+    float Voltaje=R_adc*5/1024;
+    float Rs=1000*((5-Voltaje)/Voltaje);
+//    float ppm=0.4091*pow(Rs/5463,-1.497);
+    return Rs;
+}
 
-typedef struct {
-    unsigned short adcres;
-} MQ2_t;
-
-MQ2_t MQ2;
-
-
-void ADCpropano(void) {
-    ADCON0=0x03;    //Inicia Medicion en AD0
+float valorozono(void) {
+    
+    //convercion ADC 
+    
+    ADCON0bits.CHS=1;       //seleccion de canal 1
+    ADCON0bits.ADON=1;      //encender el periferico ADC
+    __delay_us(10);          //espera requerida por el ADC
+    ADCON0bits.GO=1;        //Inicia Medicion en AD1  
     while(ADCON0bits.GO);      //Se queda esperando hasta que acabe la conversion
-    MQ2.adcres = ADRES;
+    float R_adc=ADRES;    
+    ADCON0bits.ADON=0;      //apagar el periferico ADC para ahorrar energia
+    
+    //escalamiento de las unidades
+    float Voltaje=R_adc*5/1024;
+    float Rs=1000*((5-Voltaje)/Voltaje);
+//    float ppm=0.4091*pow(Rs/5463,-1.497);
+    return Rs;
 }
 
-unsigned short valorPropano(void) {
-    ADCpropano();
-    return MQ2.adcres;
+float valorCO(void) {
+    
+    //convercion ADC 
+    
+    ADCON0bits.CHS=2;       //seleccion de canal 2
+    ADCON0bits.ADON=1;      //encender el periferico ADC
+    __delay_us(10);          //espera requerida por el ADC
+    ADCON0bits.GO=1;        //Inicia Medicion en AD2  
+    while(ADCON0bits.GO);      //Se queda esperando hasta que acabe la conversion
+    float R_adc=ADRES;
+    ADCON0bits.ADON=0;      //apagar el periferico ADC para ahorrar energia
+    
+    //escalamiento de las unidades
+    float Voltaje=R_adc*5/1024;
+    float Rs=1000*((5-Voltaje)/Voltaje);
+//    float ppm=0.4091*pow(Rs/5463,-5);
+    return Rs;
 }
+
+
+float valor135(void) {
+    
+    //convercion ADC 
+        
+    ADCON0bits.CHS=4;       //seleccion de canal 4
+    ADCON0bits.ADON=1;      //encender el periferico ADC
+    __delay_us(10);          //espera requerida por el ADC
+    ADCON0bits.GO=1;        //Inicia Medicion en AD4  
+    while(ADCON0bits.GO);      //Se queda esperando hasta que acabe la conversion
+    float R_adc=ADRES;
+    ADCON0bits.ADON=0;      //apagar el periferico ADC para ahorrar energia
+    
+    //escalamiento de las unidades
+    float Voltaje=R_adc*5/1024;
+    float Rs=1000*((5-Voltaje)/Voltaje);
+//    float ppm=0.4091*pow(Rs/5463,-5);
+    return Rs;
+}
```

Comment:
Este signo de prohibido se quita simplemente dejando una linea en blanco al final del archivo
```suggestion
}

```